mam/v2/mam/endpoint.c

```diff
@@ -165,21 +163,15 @@ size_t mam_endpoints_serialized_size(mam_endpoint_t_set_t const endpoints) {
 
 retcode_t mam_endpoints_serialize(mam_endpoint_t_set_t const endpoints,
                                   trits_t *const buffer) {
-  retcode_t ret = RC_OK;
   mam_endpoint_t_set_entry_t *entry = NULL;
   mam_endpoint_t_set_entry_t *tmp = NULL;
 
   pb3_encode_size_t(mam_endpoint_t_set_size(endpoints),
                     buffer);  // endpoints number
 
   HASH_ITER(hh, endpoints, entry, tmp) {
-    if ((ret = mam_endpoint_serialize(&entry->value, buffer)) !=
-        RC_OK) {  // endpoint
-      return ret;
-    }
+    mam_endpoint_serialize(&entry->value, buffer);
   }
-
-  return ret;
```

Comment:
This function still returns `retcode_t`

Reply:
because it uses `mam_channel_t_set_add` which allocates and can fail